src/compat/posix/proc/fork_nommu/fork.c

```diff
@@ -50,28 +50,28 @@ void _NORETURN fork_body(struct pt_regs *ptregs) {
 		panic("%s returning", __func__);
 	}
 
-	adrspc = fork_addr_space_get(parent);
-	if (!adrspc) {
-		adrspc = fork_addr_space_create(NULL);
-		fork_addr_space_set(parent, adrspc);
-	}
+	sched_lock();
+	{
+		adrspc = fork_addr_space_get(parent);
+		if (!adrspc) {
+			adrspc = fork_addr_space_create(NULL);
+			fork_addr_space_set(parent, adrspc);
+		}
 
-	/* Save the stack of the current thread */
-	fork_stack_store(adrspc, thread_self());
+		/* Save the stack of the current thread */
+		fork_stack_store(adrspc, thread_self());
 
-	child = task_table_get(child_pid);
-	child_adrspc = fork_addr_space_create(adrspc);
+		child = task_table_get(child_pid);
+		child_adrspc = fork_addr_space_create(adrspc);
 
-	/* Can't use fork_addr_space_store() as we use
-	 * different task as data source */
-	fork_stack_store(child_adrspc, child->tsk_main);
-	fork_heap_store(&child_adrspc->heap_space, task_self());
-	fork_static_store(&child_adrspc->static_space);
+		/* Can't use fork_addr_space_store() as we use
+		 * different task as data source */
+		fork_stack_store(child_adrspc, child->tsk_main);
+		fork_heap_store(&child_adrspc->heap_space, task_self());
+		fork_static_store(&child_adrspc->static_space);
 
-	memcpy(&child_adrspc->pt_entry, ptregs, sizeof(*ptregs));
+		memcpy(&child_adrspc->pt_entry, ptregs, sizeof(*ptregs));
 
-	sched_lock();
-	{
 		child = task_table_get(child_pid);
```

Comment:
This is redundant, the same at line 62.

Reply:
sched_lock is required because when fork_addr_space_create is happened  irq event may be occurred and then contex_switch in where try save data in a new addr_space which is not initialized in that place